src/Magento/FunctionalTestingFramework/StaticCheck/TestDependencyCheck.php

```diff
@@ -0,0 +1,296 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\FunctionalTestingFramework\StaticCheck;
+
+use Magento\FunctionalTestingFramework\DataGenerator\Handlers\DataObjectHandler;
+use Magento\FunctionalTestingFramework\Page\Handlers\PageObjectHandler;
+use Magento\FunctionalTestingFramework\Page\Handlers\SectionObjectHandler;
+use Magento\FunctionalTestingFramework\Test\Handlers\ActionGroupObjectHandler;
+use Magento\FunctionalTestingFramework\Test\Handlers\TestObjectHandler;
+use Magento\FunctionalTestingFramework\Config\MftfApplicationConfig;
+use Symfony\Component\Console\Input\InputInterface;
+
+/**
+ * Class TestDependencyCheck
+ * @package Magento\FunctionalTestingFramework\StaticCheck
+ */
+class TestDependencyCheck implements StaticCheckInterface
+{
+    /**
+     * Array of FullModuleName => [dependencies]
+     * @var array
+     */
+    private $allDependencies;
+
+    /**
+     * Array of FullModuleName => [dependencies], including flattened dependency tree
+     * @var array
+     */
+    private $flattenedDependencies;
+
+    /**
+     * Array of FullModuleName => PathToModule
+     * @var array
+     */
+    private $moduleNameToPath;
+
+    /**
+     * Array of FullModuleName => ComposerModuleName
+     * @var array
+     */
+    private $moduleNameToComposerName;
+
+    /**
+     * Transactional Array to keep track of what dependencies have already been extracted.
+     * @var array
+     */
+    private $alreadyExtractedDependencies;
+
+    /**
+     * Checks test dependencies, determined by references in tests versus the dependencies listed in the Magento module
+     *
+     * @param InputInterface $input
+     * @return string
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     */
+    public function execute(InputInterface $input)
+    {
+        MftfApplicationConfig::create(
+            true,
+            MftfApplicationConfig::UNIT_TEST_PHASE,
+            false,
+            false
+        );
+
+        $testObjects = TestObjectHandler::getInstance()->getAllObjects();
```

Comment:
Does it contains all test.xml files in code base or just the enabled modules from module resolver? We should scan all code base.

Reply:
The config is created with `--force`, so it's the entire code base.

Reply:
That's great!